source/_posts/2016-04-25-opencomponents-microservices-in-the-front-end-world.markdown

```diff
@@ -0,0 +1,259 @@
+
+### Another front-end framework?
+
+OpenComponents is a system for allowing people to have tools to facilitate code sharing, reduce dependencies, and easily approach new features and experiments from the back-end to the front-end. To achieve this, it is based on the concept of using services as interfaces for enabling pages to render partial contents located, executed and deployed independently.
+
+OpenComponents is not *another SPA JS framework*: it is a set of conventions, patterns and tools to develop and quickly deploy fragments of front-end. In this perspective, it plays nicely with any existing architecture and framework in terms of front-end and back-end: its purpose is to **serve as delivery mechanism for a more modularised end-result in the front-end**.
+
+OC is been in production for more than a year at OpenTable and it is [fully open-sourced][2].
+
+## Overview
+
+OpenComponents involves two parts:
+
+* The **consumers** are web pages that need fragments of html for rendering partial contents. Sometimes they need some content during server-side rendering, somethings when executing code in the browser.
+* The **components** are small units of isomorphic code mainly consisting of html, javascript, css. They can optionally contain some logic, allowing a server-side node.js closure to compose a model that is used to render the view. When rendered they are pieces of pure html, ready to be injected in any web page.
+
+The framework consists of three parts:
+
+* The **cli** allows developers to create, develop, test, and publish components.
+* The **library** is where the components are stored after the publishing. When components depend on static resources (such as images, css files, etc.) these are stored, during packaging and publishing, in a publicly-exposed part of the library that serves as cdn.
+* The **registry** is a REST API that is used to consume and publish components. It is the entity that handles the traffic between the library and the consumers.
+
+In the following example, you can see how a web page looks like when including both a server-side rendered component (*header*) and client-side (still) unrendered component (*advert*):
+
+```html
+<!DOCTYPE html>
+  ...
+  <oc-component href="//oc-registry.com/header/1.X.X" data-rendered="true">
+    <a href="/">
+      <img src="//cdn.com/oc/header/1.2.3/img/logo.png" />
+    </a>
+  </oc-component>
+  ...
+  <p>page content</p>
+  <oc-component href="//oc-registry.com/advert/~1.3.5/?type=bottom">
+  </oc-component>
+  ...
+  <script src="//oc-registry/oc-client/client.js"></script>
+```
+
+### Getting started
+
+Node.js is required for creating OC components. Then to create a component:
```

Comment:
i'd say that a little bit of text under heading would be preferable (that's what people said in uni :P and i tend to agree with that ;) )